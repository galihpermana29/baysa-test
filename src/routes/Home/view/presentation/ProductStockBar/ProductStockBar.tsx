import {
	Chart as ChartJS,
	CategoryScale,
	LinearScale,
	BarElement,
	Title,
	Tooltip,
	Legend,
} from 'chart.js';
import { Bar } from 'react-chartjs-2';

import { DemoChartI } from '../../../model/interfaces';
import CustomLegend from '../CustomLegend/CustomLegend';
import { Divider } from 'antd';

ChartJS.register(
	CategoryScale,
	LinearScale,
	BarElement,
	Title,
	Tooltip,
	Legend
);

export function DemoChart({
	customTooltip,
	config,
	xAxisLabel,
	yAxisLabel,
	headerEl,
}: DemoChartI) {
	const chartData = {
		labels: config.labels,
		datasets: [
			{
				label: 'Stock',
				data: config.dataset,
				backgroundColor: 'rgba(75,192,192,0.6)',
				...config.styles,
			},
		],
	};

	const chartOptions: any = {
		scales: {
			x: {
				title: {
					display: true,
					text: xAxisLabel,
				},
			},
			y: {
				beginAtZero: true,
				title: {
					display: true,
					text: yAxisLabel,
				},
			},
		},
		plugins: {
			legend: {
				display: false,
			},
			tooltip: {
				enabled: false,
				...(customTooltip !== null
					? { external: (context: any) => customTooltip!(context, config.data) }
					: {}),
			},
		},
		...config,
	};

	return (
		<div>
			{headerEl}
			<Divider />
			<Bar data={chartData} options={chartOptions} />
			<CustomLegend data={config.labels} />
		</div>
	);
}
