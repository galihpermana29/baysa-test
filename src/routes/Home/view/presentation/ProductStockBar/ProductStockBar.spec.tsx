import { render, screen } from '@testing-library/react';
import { DemoChart } from './ProductStockBar';

const mockConfig = {
	labels: ['Label 1', 'Label 2', 'Label 3'],
	dataset: [10, 20, 30],
	styles: {
		backgroundColor: ['red', 'green', 'blue'],
	},
	data: [],
};

const mockXAxis = 'Product';
const mockYAxis = 'Stock';

test('renders DemoChartBar component with provided configuration and dataset', () => {
	const wrapper = render(
		<DemoChart
			xAxisLabel={mockXAxis}
			yAxisLabel={mockYAxis}
			config={mockConfig}
			headerEl={<div>header el</div>}
		/>
	);

	const headerEl = wrapper.getByText('header el');

	mockConfig.labels.forEach((data) => {
		const legendTextEl = screen.getByText(data);
		expect(legendTextEl).toBeInTheDocument();
	});
	expect(headerEl).toBeInTheDocument();
	expect(wrapper).toMatchSnapshot();
});
