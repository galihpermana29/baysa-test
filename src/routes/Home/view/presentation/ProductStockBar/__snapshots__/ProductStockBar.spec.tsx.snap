// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DemoChartBar component with provided configuration and dataset 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          header el
        </div>
        <div
          class="ant-divider css-dev-only-do-not-override-1qhpsh8 ant-divider-horizontal"
          role="separator"
        />
        <div
          class="flex max-w-[800px] overflow-x-scroll gap-[10px] mt-[20px]"
        >
          <div
            class="flex items-center gap-[10px] bg-slate-50 min-w-max p-[10px] rounded-lg"
          >
            <div
              class="w-[10px] h-[10px] rounded-full bg-black"
            />
            <div
              class="flex-1"
            >
              Label 1
            </div>
          </div>
          <div
            class="flex items-center gap-[10px] bg-slate-50 min-w-max p-[10px] rounded-lg"
          >
            <div
              class="w-[10px] h-[10px] rounded-full bg-black"
            />
            <div
              class="flex-1"
            >
              Label 2
            </div>
          </div>
          <div
            class="flex items-center gap-[10px] bg-slate-50 min-w-max p-[10px] rounded-lg"
          >
            <div
              class="w-[10px] h-[10px] rounded-full bg-black"
            />
            <div
              class="flex-1"
            >
              Label 3
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        header el
      </div>
      <div
        class="ant-divider css-dev-only-do-not-override-1qhpsh8 ant-divider-horizontal"
        role="separator"
      />
      <div
        class="flex max-w-[800px] overflow-x-scroll gap-[10px] mt-[20px]"
      >
        <div
          class="flex items-center gap-[10px] bg-slate-50 min-w-max p-[10px] rounded-lg"
        >
          <div
            class="w-[10px] h-[10px] rounded-full bg-black"
          />
          <div
            class="flex-1"
          >
            Label 1
          </div>
        </div>
        <div
          class="flex items-center gap-[10px] bg-slate-50 min-w-max p-[10px] rounded-lg"
        >
          <div
            class="w-[10px] h-[10px] rounded-full bg-black"
          />
          <div
            class="flex-1"
          >
            Label 2
          </div>
        </div>
        <div
          class="flex items-center gap-[10px] bg-slate-50 min-w-max p-[10px] rounded-lg"
        >
          <div
            class="w-[10px] h-[10px] rounded-full bg-black"
          />
          <div
            class="flex-1"
          >
            Label 3
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
